config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/EncassEntityBuilder.java

```diff
@@ -82,21 +83,59 @@ private boolean isAnnotatedEntity(Encass encass, AnnotatedEntity annotatedEntity
                 EntityTypes.ENCAPSULATED_ASSERTION_TYPE.equals(annotatedEntity.getEntityType());
     }
 
-    private Entity buildEncassEntity(AnnotatedBundle annotatedBundle, Bundle bundle, String name, Encass encass, Document document) {
-        Policy policy = bundle.getPolicies().get(encass.getPolicy());
+    private String getPolicyId(String policyWithPath, Bundle bundle) {
+        Policy policy = bundle.getPolicies().get(policyWithPath);
+        String policyId = null;
         if (policy == null) {
-            throw new EntityBuilderException("Could not find policy for encass. Policy Path: " + encass.getPolicy());
+            Set<BundleDefinedEntities> bundleMetadataSet = bundle.getMetadataDependencyBundles();
+            if (bundleMetadataSet != null && !bundleMetadataSet.isEmpty()) {
+                for (BundleDefinedEntities bundleMetadata : bundleMetadataSet) {
+                    Collection<BundleDefinedEntities.DefaultMetadata> metadataCollection = bundleMetadata.getDefinedEntities();
+                    if (metadataCollection != null) {
+                        Optional<BundleDefinedEntities.DefaultMetadata> optionalMetadata = metadataCollection.stream().
+                                filter(metadata -> metadata.getName().equals(PathUtils.extractName(policyWithPath))
+                                        && metadata.getType().equals(EntityTypes.POLICY_TYPE)).findFirst();
+                        if (optionalMetadata.isPresent()) {
+                            if (policyId == null) {
+                                policyId = optionalMetadata.get().getId();
```

Comment:
is this logic is not applicable to set GUID?

Reply:
correct me , this is to find the dependent policy from other modules?

Reply:
yes